src/cdk/drag-drop/drop.ts

```diff
@@ -217,9 +220,17 @@ export class CdkDrop<T = any> implements OnInit, OnDestroy {
    * @param item Item whose index should be determined.
    */
   getItemIndex(item: CdkDrag): number {
-    return this._dragging ?
-        findIndex(this._positionCache.items, currentItem => currentItem.drag === item) :
-        this._draggables.toArray().indexOf(item);
+    if (!this._dragging) {
+      return this._draggables.toArray().indexOf(item);
+    }
+
+    // Items are sorted always by top/left in the cache, however they flow differently in RTL.
+    // The rest of the logic still stands no matter what orientation we're in, however
+    // we need to invert the array when determining the index.
+    const items = this.orientation === 'horizontal' && this._dir && this._dir.value === 'rtl' ?
+        this._positionCache.items.slice().reverse() : this._positionCache.items;
```

Comment:
How often is `getItemIndex` called? I'm wondering if `slice().reverse()` might get expensive for large lists

Reply:
It's called twice when an item is dropped to figure out its old index and the new one. I don't imagine it getting too expensive, unless we're dealing with a massive array.